irods/library/tests/test_irods_permission.yml

```diff
@@ -62,6 +63,20 @@
         - "{{ sub_data_exclusive }}"
       tags: non_idempotent
 
+    - name: Create a large, nested collection structure
+      ansible.builtin.shell:
+        executable: /bin/bash
+        cmd: |
+          for i in {1..100}; do
+            imkdir -p "{{ coll_deep }}/level_$i"
+            for j in {1..100}; do
+              file="$(mktemp)"
+              iput -R ingestRes "$file" "{{ coll_deep }}/level_$i/file_$j"
+            done
+          done
```

Comment:
I'm not asking you to fix anything here. This is just a tip on performance.

It is faster to generate the directory structure first and then use a single bulk iput to upload it into iRODS. For example, the following took a little over three minutes to run on my computer.

```console
for i in {1..100}; do
  mkdir -p "dir_deep/level_$i"
  for j in {1..100}; do
    touch dir_deep/level_"$i"/file_"$j"
  done
done
iput -b -I -r -R ingestRes dir_deep "{{ coll_deep }}"
```